config/muxer.go

```diff
@@ -43,21 +42,23 @@ func MuxerConstructor(m interface{}) (MuxC, error) {
 	}, nil
 }
 
-func makeMuxer(h host.Host, tpts []MsMuxC) (network.Multiplexer, error) {
-	muxMuxer := msmux.NewBlankTransport()
+func makeMuxers(h host.Host, tpts []MsMuxC) (map[string]network.Multiplexer, []string, error) {
 	transportSet := make(map[string]struct{}, len(tpts))
+	muxers := make(map[string]network.Multiplexer)
+	preference := make([]string, 0)
```

Comment:
Preallocate slices with the correct capacity, if you know the capacity in advance.